src/NServiceBus.Core/Serialization/SerializationSettingsExtensions.cs

```diff
@@ -38,6 +40,8 @@ public static void SetMainSerializer(this SettingsHolder settings, Serialization
         {
             if (!settings.TryGet(MainSerializerSettingsKey, out Tuple<SerializationDefinition, SettingsHolder> defaultSerializerAndSettings))
             {
+                var noDefaultSerializerMsg = $"Because no message serializer was selected, the default {nameof(XmlSerializer)} will be used instead. In a future version of NServiceBus the {nameof(XmlSerializer)} will no longer be the default. For better forward compatibility, either choose a different message serializer, or make the choice of XML serialization explicit using endpointConfiguration.{nameof(SerializationConfigExtensions.UseSerialization)}<{nameof(XmlSerializer)}>()";
```

Comment:
Note to self: We need to go through and update samples to get rid of the warning